### ISSUE TITLE
Can't get %p (or %P) to work - what am I doing wrong?

### ISSUE THREAD

**Issue by hepcat72** [houseabsolute/DateTime-Format-Strptime#31]
```
perl -e '$d="6/15/21, 6:00 PM";use DateTime::Format::Strptime;print("DateTime::Format::Strptime::VERSION: ",$DateTime::Format::Strptime::VERSION,"\n");my $strp = DateTime::Format::Strptime->new(pattern   => "%D, %R",locale    => "en_US",time_zone => "America/New_York");$r=$strp->parse_datetime($d);print((defined($r) ? "$r" : "undef"),"\n");$strp = DateTime::Format::Strptime->new(pattern   => "%D, %R %p",locale    => "en_US",time_zone => "America/New_York");$r=$strp->parse_datetime($d);print((defined($r) ? "$r" : "undef"),"\n")'
DateTime::Format::Strptime::VERSION: 1.78
2021-06-15T06:00:00
undef
```
I have tried all sorts of variations, but in no case can I seem to get `%p` to match "PM".

Formatted version:
```
use DateTime::Format::Strptime;

$d="6/15/21, 6:00 PM";

print("DateTime::Format::Strptime::VERSION: ",
      $DateTime::Format::Strptime::VERSION,"\n");

my $strp = DateTime::Format::Strptime->new(pattern   => "%D, %R",
                                           locale    => "en_US",
                                           time_zone => "America/New_York");

$r=$strp->parse_datetime($d);

print((defined($r) ? "$r" : "undef"),"\n");

$strp = DateTime::Format::Strptime->new(pattern   => "%D, %R %p",
                                        locale    => "en_US",
                                        time_zone => "America/New_York");

$r=$strp->parse_datetime($d);

print((defined($r) ? "$r" : "undef"),"\n");
```

**Comment by autarch** [comment]
This one liner is really hard to follow. Can you redo this as a multi-line program?

**Comment by hepcat72** [comment]
OK, how's that (above)?

**Comment by hepcat72** [comment]
Sorry, I'm used to ridiculous one-liners.

**Comment by hepcat72** [comment]
Currently, I worked around my inability to match `%p` (or `%P`), but doing something like this:

```
my $pattern = '%D, %R AM';

my $strp = DateTime::Format::Strptime->new(pattern   => $pattern,
                                           locale    => "en_US",
                                           time_zone => "America/New_York");

$r=$strp->parse_datetime($d);

if(!defined($r))
  {
    $pattern = '%D, %R PM';
    $strp = DateTime::Format::Strptime->new(pattern   => $pattern,
                                               locale    => "en_US",
                                               time_zone => "America/New_York");

    $r=$strp->parse_datetime($d);
    if(!defined($r))
      {die "Parsing error"}
    $r->add(hours => 12);
  }
```

**Comment by autarch** [comment]
> OK, how's that (above)?

Perfect. Thanks!

**Comment by autarch** [comment]
I think I see what's going on. The `%R` pattern is equivalent to `%H:%M`. Note that capital "H" there. The `%H` pattern matches an hour from 00-23 (and works with single digits as well). That means that the parser will ignore the am/pm piece, because it already has a valid hour. If you replace `%R` with `%I:%M` then it will match a 12 hour time (1-12) and use the am/pm field to determine the correct hour.

**Comment by hepcat72** [comment]
Thanks!  I will update my code!
